### PR TITLE
swagger v2 tag feature implemented

### PR DESCRIPTION
**Pull request by claymodel**
Name-spcae based grouping in swagger doc V2.0

Before Change
grape-swagger-v2-before

After Change
grape-swagger-v2-new

After Change Corrected
grape-swagger-v2-new-corrected

### CODE REVIEW

**Comment by claymodel** [comment]
updated and corrected the grouping

**Comment by dblock** [comment]
This still needs to pass the build and be squashed, please.

File: lib/grape-swagger/endpoint.rb [hunk 1]
```diff
@@ -28,7 +28,7 @@ def content_types_for(target_class)
     # swagger spec2.0 related parts
     #
     # required keys for SwaggerObject
-    def swagger_object(info, target_class, request, options)
+    def swagger_object(info, target_class, request, options, namespace_routes_array)
```
Comment (dblock) [inline lib/grape-swagger/endpoint.rb]: Why wound't the `namespace_routes_array` be passed with `options`?

File: lib/grape-swagger/endpoint.rb [hunk 2]
```diff
@@ -132,6 +133,7 @@ def method_object(route, options)
       methods[:produces] = produces_object(route, options)
 
       methods[:parameters] = params_object(route)
+      methods[:tags]      = Array(route.route_path.split('{')[0].split('/').delete_if { |i| (i==route.route_prefix) || i[/v[0-9]*/] }.reject(&:empty?).first)
```
Comment (dblock) [inline lib/grape-swagger/endpoint.rb]: Should be refactored out.

File: lib/grape-swagger/doc_methods.rb [hunk 3]
```diff
@@ -34,11 +34,28 @@ def setup(options)
         header['Access-Control-Allow-Origin']   = '*'
         header['Access-Control-Request-Method'] = '*'
 
+        namespaces = target_class.combined_namespaces
+        namespace_routes = target_class.combined_namespace_routes
+
+        namespace_routes_array = namespace_routes.keys.map do |local_route|
+          next if namespace_routes[local_route].map(&:route_hidden).all? { |value| value.respond_to?(:call) ? value.call : value }
+
+          original_namespace_name = target_class.combined_namespace_identifiers.key?(local_route) ? target_class.combined_namespace_identifiers[local_route] : local_route
```
Comment (dblock) [inline lib/grape-swagger/doc_methods.rb]: Try to refactor the whole `description` logic out of here.

Comment (claymodel) [inline lib/grape-swagger/doc_methods.rb]: Hi Daniel, Thank you for the feedback.
https://github.com/ruby-grape/grape-swagger/pull/348
I have updated the code and the build is passing now.
looking forward for your feed back.

**Pull request closed without merging**